spacehammer/core.fnl

```diff
@@ -73,133 +49,41 @@ Returns nil. This function causes side-effects.
           "
           (error "get-config can only be called after all modules have initialized")))
 
-(fn file-exists?
-  [filepath]
-  "
-  Determine if a file exists and is readable.
-  Takes a file path string
-  Returns true if file is readable
-  "
-  (let [file (io.open filepath "r")]
-    (when file
-      (io.close file))
-    (~= file nil)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;; create custom config file if it doesn't exist
+;; create custom config and/or init files if either doesn't exist
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(fn copy-file
-  [source dest]
-  "
-  Copies the contents of a source file to a destination file.
-  Takes a source file path and a destination file path.
-  Returns nil
-  "
-  (let [default-config (io.open source "r")
-        custom-config (io.open dest "a")]
-    (each [line _ (: default-config :lines)]
-      (: custom-config :write (.. line "\n")))
-    (: custom-config :close)
-    (: default-config :close)))
 
 ;; If ~/.spacehammer/config.fnl does not exist
 ;; - Create ~/.spacehammer dir
 ;; - Copy default config.example.fnl to ~/.spacehammer/config.fnl
-(let [example-path (hs.spoons.resourcePath "config.example.fnl")
-      target-path (.. customdir "/config.fnl")]
-  (when (not (file-exists? target-path))
-    (log.d (.. "Copying " example-path " to " target-path))
+;; If ~/.spacehammer/init.fnl does not exist
+;; - Copy default init.example.fnl to ~/.spacehammer/init.fnl
+(let [example-config-path (hs.spoons.resourcePath "config.example.fnl")
+      example-init-path (hs.spoons.resourcePath "init.example.fnl")
+      target-config-path (.. customdir "/config.fnl")
+      target-init-path (.. customdir "/init.fnl")]
+  (when (not (file-exists? target-config-path))
```

Comment:
there's an argument for pulling the copy-pasta logic here into a local helper function, but that's more an aesthetic question than a practical maintenance one: I don't think there's much inside these `when` forms that's likely to change unless the whole "copy example file if missing" approach is changed.

Reply:
This looks good to me, and we can always change it later.

Reply:
Most of the time I prefer to leave things duplicated when it's 2 - 3 times, it can always be cleaned up later and doing so is not likely to affect userland at all.

One thought I had was wrapping this in a function of the Spoon, like `Spacehammer:generateDefaultConfigIfMissing()` and including it in our setup instructions for users to paste in. It should work the same way, but if a user such as myself knows I'm going to have my own config then that line could be commented out or removed to avoid those extra FS checks.

But then again it's not that expensive. Either way, can always change this later.

Reply:
I think it makes sense to keep the checks; as you said, they're cheap, and if they are opt-in, we have to solve the problem of what to do if there are no default configs and the user missed the note to call that function.

If we later decide the efficiency gainz are worth a little extra complexity, I'd prefer we add fuss to the power user use case than the beginner one, e.g.:
```lua
local Spacehammer = hs.loadSpoon("Spacehammer")
Spacehammer.suppress_config_check = true
Spacehammer:start()
```

Reply:
Good point